Content.Shared/Chemistry/EntitySystems/SharedHypospraySystem.cs

```diff
@@ -0,0 +1,74 @@
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Timing;
+using Content.Shared.Verbs;
+using Content.Shared.Popups;
+using Robust.Shared.Player;
+using Content.Shared.Administration.Logs;
+
+namespace Content.Shared.Chemistry.EntitySystems;
+
+public abstract class SharedHypospraySystem : EntitySystem
+{
+    [Dependency] protected readonly UseDelaySystem _useDelay = default!;
+    [Dependency] protected readonly SharedPopupSystem _popup = default!;
+    [Dependency] protected readonly SharedSolutionContainerSystem _solutionContainers = default!;
+    [Dependency] protected readonly IEntityManager _entMan = default!;
+    [Dependency] protected readonly ISharedAdminLogManager _adminLogger = default!;
+    [Dependency] protected readonly ReactiveSystem _reactiveSystem = default!;
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<HyposprayComponent, GetVerbsEvent<AlternativeVerb>>(AddToggleModeVerb);
+    }
+
+    //
+    // Uses the OnlyMobs field as a check to implement the ability
+    // to draw from jugs and containers with the hypospray
+    // Toggleable to allow people to inject containers if they prefer it over drawing
+    //
+    private void AddToggleModeVerb(Entity<HyposprayComponent> entity, ref GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract || args.Hands == null)
+            return;
+
+        if (!HasComp<ActorComponent>(args.User))
+            return;
+
+        var (_, component) = entity;
+        var user = args.User;
+        var verb = new AlternativeVerb
+        {
+            Text = Loc.GetString("hypospray-verb-mode-label"),
+            Act = () =>
+            {
+                ToggleMode(entity, user);
+            }
+        };
+        args.Verbs.Add(verb);
+    }
+
+    private void ToggleMode(Entity<HyposprayComponent> entity, EntityUid user)
+    {
+        string msg;
+        switch (entity.Comp.ToggleMode)
+        {
+            case HyposprayToggleMode.All:
+                SetMode(entity, HyposprayToggleMode.OnlyMobs);
+                msg = "hypospray-verb-mode-inject-mobs-only";
+                break;
+            case HyposprayToggleMode.OnlyMobs:
+                SetMode(entity, HyposprayToggleMode.All);
+                msg = "hypospray-verb-mode-inject-all";
+                break;
+            default:
+                throw new ArgumentOutOfRangeException();
+        }
+
+        _popup.PopupEntity(Loc.GetString(msg), entity, user);
+    }
```

Comment:
PopupClient.

Reply:
My main intention of moving it to Shared was to use PopupClient but when I tried it in testing locally, the message never seemed to pop up for me?

Reply:
After going through the other changes, can confirm that PopupClient still doesn't cause the message to appear for hyposprays, but it does work fine for injectors

Mmmm....

Reply:
EmoGarbage let me know that it was due to me not having a HypospraySystem file in Client, forgot that I would need one because the HyposprayStatusControls were in the InjectorSystem file in Client. I moved that over and now this works, thanks!